kotlin-inject-compiler-core/src/main/kotlin/me/tatarka/inject/compiler/AstVisibility.kt

```diff
@@ -0,0 +1,16 @@
+package me.tatarka.inject.compiler
+
+import com.squareup.kotlinpoet.KModifier
+
+enum class AstVisibility { PUBLIC, PRIVATE, PROTECTED, INTERNAL }
```

Comment:
I actually had started with a modifier enum but switched to methods like `isPrivate` because it felt simpler and matched the ksp api.

Reply:
I think having a state which represents which visibility an element well since it can only be 1 of these states at a time, and never more than one which this enum makes clear. Perhaps adding methods to the Visibility interface to check if it's public, private, internal or protected would help?